modules/ssh/ssh.go

```diff
@@ -400,12 +406,21 @@ func Listen(host string, port int, ciphers, keyExchanges, macs []string) {
 // Public key is encoded in the format for inclusion in an OpenSSH authorized_keys file.
 // Private Key generated is PEM encoded
 func GenKeyPair(keyPath string) error {
-	privateKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	bits := 4096
+	keytype := filepath.Ext(keyPath)
```

Comment:
It doesn't seem right to guess the content by file extension. The key itself has the correct information.

```
$ ssh-keygen -t ecdsa -f a
...
$ ssh-keygen -lf a
256 SHA256:NS1YmfwZDYoEgtS3Ne0wRuDvGFFWyQfc1QB8ZT2SGeA xiaoguang@Xiaoguangs-MacBook-Pro.local (ECDSA)
```